app/src/main/java/ch/epfl/sweng/favors/favors/FavorListAdapter.java

```diff
@@ -49,14 +57,36 @@ public void onClick(View v) {}
 
         public void bind(final Favor item, final OnItemClickListener listener){
             Favor favor = item;
+            setTitleAndDescription(favor);
+            setTimestamp(favor);
+            setLocation(favor);
+            itemView.setOnClickListener(v -> listener.onItemClick(item));
+        }
+
+        private void setLocation(Favor favor) {
+            if(favor.get(Favor.StringFields.locationCity) != null)
+                location.setText(favor.get(Favor.StringFields.locationCity));
+            if(favor.get(Favor.ObjectFields.location) != null) {
+                ObservableField<Object> geo = favor.getObservableObject(Favor.ObjectFields.location);
+                distance.setText(LocationHandler.distanceBetween((GeoPoint)geo.get()));
+            } else { distance.setText("--"); }
+        }
+
+        private void setTitleAndDescription(Favor favor) {
             if(favor.get(Favor.StringFields.title) != null)
                 title.setText(favor.get(Favor.StringFields.title));
-            if(favor.get(Favor.IntegerFields.creationTimestamp) != null)
-                timestamp.setText(favor.get(Favor.IntegerFields.creationTimestamp));
             if(favor.get(Favor.StringFields.description) != null)
                 description.setText(favor.get(Favor.StringFields.description));
-            itemView.setOnClickListener(v -> listener.onItemClick(item));
+            setLocation(favor);
```

Comment:
This should not be here or?
Feedback @Jmion @jeniffer9 @Charrlyine please [you are the reviewers]
I think I did an error when refactoring and I am to blame for this line.
Just give me a heads-up if I can remove it